Lifetime errors with skipped `&'static` variable

I needed to evolve my format, so I went from (simplified)
```rust
#[derive(Debug, PartialEq)]
#[derive(Serialize, Deserialize)]
#[serde(deny_unknown_fields)]
pub struct SassOptions {
    #[serde(skip)]
    pub import_dir: &'static str,
    pub style: SassOutputStyle,
}

impl Default for SassOptions {
    fn default() -> SassOptions {
        ...
    }
}

#[derive(Debug, PartialEq)]
#[derive(Serialize, Deserialize)]
#[serde(deny_unknown_fields, default)]
pub struct ConfigBuilder {
    #[serde(skip)]
    pub root: path::PathBuf,
    pub source: String,
    pub destination: String,
    pub sass: sass::SassOptions,
}
```
to
```rust

```rust
#[derive(Debug, PartialEq)]
#[derive(Serialize, Deserialize)]
#[serde(deny_unknown_fields)]
pub struct SassOptions {
    #[serde(skip)]
    pub import_dir: &'static str,
    pub style: SassOutputStyle,
}

impl Default for SassOptions {
    fn default() -> SassOptions {
        ...
    }
}

#[derive(Debug, PartialEq, Default)] 
#[derive(Serialize, Deserialize)]    
#[serde(deny_unknown_fields)]        
pub struct AssetsBuilder {           
    pub sass: sass::SassOptions,     
}                                    

#[derive(Debug, PartialEq)]
#[derive(Serialize, Deserialize)]
#[serde(deny_unknown_fields, default)]
pub struct ConfigBuilder {
    #[serde(skip)]
    pub root: path::PathBuf,
    pub source: String,
    pub destination: String,
    pub assets: AssetsBuilder,
}
```

and I started getting lifetime errors
```
error[E0495]: cannot infer an appropriate lifetime for lifetime parameter `'de` due to conflicting requirements
  --> src/config.rs:95:21
   |
95 | #[derive(Serialize, Deserialize)]
   |                     ^^^^^^^^^^^
   |
note: first, the lifetime cannot outlive the lifetime 'de as defined on the impl at 95:21...
  --> src/config.rs:95:21
   |
95 | #[derive(Serialize, Deserialize)]
   |                     ^^^^^^^^^^^
note: ...so that types are compatible (expected std::convert::From<<__A as serde::de::MapAccess<'_>>::Error>, found std::convert::From<<__A as serde::de::MapAccess<'de>>::Error>)
  --> src/config.rs:95:21
   |
95 | #[derive(Serialize, Deserialize)]
   |                     ^^^^^^^^^^^
   = note: but, the lifetime must be valid for the static lifetime...
note: ...so that types are compatible (expected serde::Deserialize<'_>, found serde::Deserialize<'static>)
  --> src/config.rs:95:21
   |
95 | #[derive(Serialize, Deserialize)]
   |                     ^^^^^^^^^^^
   = note: this error originates in a macro outside of the current crate

error: aborting due to 2 previous errors

error: Could not compile `cobalt-bin`.
```

Situations
- Adding a layer caused the warning
- Switching `import_dir` to `String` removed the error
- Adding a member to `AssetsBuilder` did not fix things.

Original source: https://github.com/cobalt-org/cobalt.rs
Broken source: not uploaded at the moment

--- a/serde_derive/src/de.rs
+++ b/serde_derive/src/de.rs
@@ -203,7 +203,9 @@ impl BorrowedLifetimes {
 fn borrowed_lifetimes(cont: &Container) -> BorrowedLifetimes {
     let mut lifetimes = BTreeSet::new();
     for field in cont.body.all_fields() {
-        lifetimes.extend(field.attrs.borrowed_lifetimes().iter().cloned());
+        if !field.attrs.skip_deserializing() {
+            lifetimes.extend(field.attrs.borrowed_lifetimes().iter().cloned());
+        }
     }
     if lifetimes.iter().any(|b| b.ident == "'static") {
         BorrowedLifetimes::Static

--- a/test_suite/tests/test_gen.rs
+++ b/test_suite/tests/test_gen.rs
@@ -511,6 +511,14 @@ fn test_gen() {
         Tuple(&'a str, &'static str),
         Newtype(&'static str),
     }
+
+    #[derive(Serialize, Deserialize)]
+    struct SkippedStaticStr {
+        #[serde(skip_deserializing)]
+        skipped: &'static str,
+        other: isize,
+    }
+    assert::<SkippedStaticStr>();
 }
 
 //////////////////////////////////////////////////////////////////////////

